src/main/java/gregtech/api/fluids/FluidBuilder.java

```diff
@@ -0,0 +1,415 @@
+
+    /**
+     * Disables coloring the fluid. A color should still be specified.
+     *
+     * @return this
+     */
+    public @NotNull FluidBuilder disableColor() {
+        this.isColorEnabled = false;
+        return this;
+    }
+
+    /**
+     * @param mcDensity the density in MC's units
+     * @return this
+     */
+    public @NotNull FluidBuilder density(int mcDensity) {
+        this.density = mcDensity;
+        return this;
+    }
+
+    /**
+     * @param density the density in g/cm^3
+     * @return this
+     */
+    public @NotNull FluidBuilder density(double density) {
+        return density(convertToMCDensity(density));
+    }
+
+    /**
+     * Converts a density value in g/cm^3 to an MC fluid density by comparison to air's density.
+     * @param density the density to convert
+     * @return the MC integer density
+     */
+    private static int convertToMCDensity(double density) {
+        // conversion formula from GT6
+        if (density > 0.001225) {
+            return (int) (1000 * density);
+        } else if (density < 0.001225) {
+            return (int) (-0.1 / density);
+        }
+        return 0;
+    }
+
+    /**
+     * @param luminosity of the fluid from [0, 16)
+     * @return this
+     */
+    public @NotNull FluidBuilder luminosity(int luminosity) {
+        Preconditions.checkArgument(luminosity >= 0 && luminosity < 16, "luminosity must be >= 0 and < 16");
+        this.luminosity = luminosity;
+        return this;
+    }
+
+    /**
+     * @param mcViscosity the MC viscosity of the fluid
+     * @return this
+     */
+    public @NotNull FluidBuilder viscosity(int mcViscosity) {
+        Preconditions.checkArgument(mcViscosity >= 0, "viscosity must be >= 0");
+        this.viscosity = mcViscosity;
+        return this;
+    }
+
+    /**
+     * @param viscosity the viscosity of the fluid in Poise
+     * @return this
+     */
+    public @NotNull FluidBuilder viscosity(double viscosity) {
+        return viscosity(convertViscosity(viscosity));
+    }
+
+    /**
+     * Converts viscosity in Poise to MC viscosity
+     * @param viscosity the viscosity to convert
+     * @return the converted value
+     */
+    private static int convertViscosity(double viscosity) {
+        return (int) (viscosity * 10000);
+    }
+
+    /**
+     * @param attribute the attribute to add
+     * @return this
+     */
+    public @NotNull FluidBuilder attribute(@NotNull FluidAttribute attribute) {
+        this.attributes.add(attribute);
+        return this;
+    }
+
+    /**
+     * @param attributes the attributes to add
+     * @return this
+     */
+    public @NotNull FluidBuilder attributes(@NotNull FluidAttribute @NotNull ... attributes) {
+        Collections.addAll(this.attributes, attributes);
+        return this;
+    }
+
+    /**
+     * Mark this fluid as having a custom still texture
+     * @return this
+     */
+    public @NotNull FluidBuilder customStill() {
+        return textures(true);
```

Comment:
What is the point of this redirect? `textures` is only called from this method. It should be fine to just merge the contents of the `textures` method into this method